godtools/Views/TractElements/TractElementFunctions.swift

```diff
@@ -84,4 +85,17 @@ extension BaseTractElement {
         return (elementName!, namespace, kind, properties, children)
     }
     
+    func elementListener() -> String {
+        return ""
+    }
+    
+    func sendMessageToElement(tag: String) {
```

Comment:
Here's some of the code. (it's kind of verbose), but basically what we did, is for any listener that's defined on a page or a modal, we look for notifications that have that event as a key.

https://github.com/CruGlobal/GTViewController/blob/master/GTViewController/Classes/Nav%20Classes/GTViewController.m#L590